api/azure/utils.go

```diff
@@ -848,23 +848,40 @@ func (config *AzureProvider) ConfigReader(logging log.Logger, clusterType string
 	return nil
 }
 
+// checking if the environment variables are set or not.
 func setRequiredENV_VAR(logging log.Logger, ctx context.Context, cred *AzureProvider) error {
 	tokens, err := util.GetCred(logging, "azure")
 	if err != nil {
 		return err
 	}
 	cred.SubscriptionID = tokens["subscription_id"]
```

Comment:
if the file is not present then this line will through panic
as token will not have the entries

### my suggestion
first make it accept from env var then if environment variable is not present then add token[..]